### PR TITLE
glossary-push/pull for zanata-cli

### PR DESCRIPTION
**Pull request by aeng**
https://zanata.atlassian.net/browse/ZNTA-44

---
This change is [Reviewable](https://reviewable.io/reviews/zanata/zanata-client/120)

### CODE REVIEW

**Comment by huangp** [comment]
Reviewed 12 of 13 files at r1.
Review status: 12 of 13 files reviewed at latest revision, 4 unresolved discussions.

---

*[docs/commands/glossary-pull.md, line 13 \[r1\]](https://reviewable.io:443/reviews/zanata/zanata-client/120#-KLFPsESp6KXMFPQ4GuI:-KLFPsESp6KXMFPQ4GuJ:2050088014) ([raw file](https://github.com/zanata/zanata-client/blob/87cc292ababddcf1efd9fe6d5aa607843dd651df/docs/commands/glossary-pull.md#L13)):*
> ```Markdown
> 3. The file will be in `.csv` format. (default)
> 
> To download different as format, use the `--file-type` options (csv or po). 
> ```

To download  in different format?

---

*[docs/commands/glossary-pull.md, line 20 \[r1\]](https://reviewable.io:443/reviews/zanata/zanata-client/120#-KLFPym6zxi-F1o3eeob:-KLFPym6zxi-F1o3eeoc:-1617620482) ([raw file](https://github.com/zanata/zanata-client/blob/87cc292ababddcf1efd9fe6d5aa607843dd651df/docs/commands/glossary-pull.md#L20)):*
> ```Markdown
> ```
> 
> To download only few locales for `--file-type po`, use the ` --trans-lang` options. 
> ```

To download only specific locales for ...

---

*[zanata-client-commands/src/main/java/org/zanata/client/commands/glossary/delete/GlossaryDeleteOptionsImpl.java, line 30 \[r1\]](https://reviewable.io:443/reviews/zanata/zanata-client/120#-KLFQS0CO9vsel1lFA5Q:-KLFQS0D69SZPSlJ2Nyi:-1315727814) ([raw file](https://github.com/zanata/zanata-client/blob/87cc292ababddcf1efd9fe6d5aa607843dd651df/zanata-client-commands/src/main/java/org/zanata/client/commands/glossary/delete/GlossaryDeleteOptionsImpl.java#L30)):*
> ```Java
> 
>     @Option(name = "--all", metaVar = "ALL",
>         usage = "Delete entire glossaries in server. Default: false")
> ```

Delete entire glossary from the server?

---

*[zanata-client-commands/src/main/java/org/zanata/client/commands/glossary/delete/GlossaryDeleteOptionsImpl.java, line 47 \[r1\]](https://reviewable.io:443/reviews/zanata/zanata-client/120#-KLFQaBNIjzNVqu0J0FG:-KLFQaBNIjzNVqu0J0FH:94744209) ([raw file](https://github.com/zanata/zanata-client/blob/87cc292ababddcf1efd9fe6d5aa607843dd651df/zanata-client-commands/src/main/java/org/zanata/client/commands/glossary/delete/GlossaryDeleteOptionsImpl.java#L47)):*
> ```Java
>     @Override
>     public String getCommandDescription() {
>         return "Delete glossary entry in Zanata";
> ```

entry(s)

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-client/120)*

**Comment by huangp** [comment]
:lgtm: -(Awaiting review) +Reviewed +(On QA) 

---

Reviewed 2 of 4 files at r2.
Review status: 12 of 14 files reviewed at latest revision, 4 unresolved discussions.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-client/120#-:-KLFUtuuABLw92XDPrgj:7167892)*

**Comment by seanf** [comment]
*[zanata-client-commands/src/main/java/org/zanata/client/commands/glossary/delete/GlossaryDeleteOptionsImpl.java, line 47 \[r1\]](https://reviewable.io:443/reviews/zanata/zanata-client/120#-KLFQaBNIjzNVqu0J0FG:-KLFc8U0Yr94vuK1LpKp:1351717696) ([raw file](https://github.com/zanata/zanata-client/blob/87cc292ababddcf1efd9fe6d5aa607843dd651df/zanata-client-commands/src/main/java/org/zanata/client/commands/glossary/delete/GlossaryDeleteOptionsImpl.java#L47)):*
<details><summary><i>Previously, huangp (Patrick Huang) wrote…</i></summary>
> entry(s)
</details>
Except that the plural of entry is entries, not entrys...

I guess we could have "entry/entries", maybe.

Can we just use the plural by itself here?

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/zanata/zanata-client/120)*

File: docs/commands/glossary-delete.md [hunk 1]
```diff
@@ -0,0 +1,21 @@
+To push delete glossary entry in Zanata, the command-line client's `glossary-delete` command can be used.
```
Comment (djansen-redhat) [inline docs/commands/glossary-delete.md]: Is "push delete glossary entry" correct terminology here?

File: zanata-client-commands/src/main/java/org/zanata/client/commands/glossary/pull/GlossaryPullOptionsImpl.java [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+ */
+public class GlossaryPullOptionsImpl extends ConfigurableGlossaryOptionsImpl
+        implements GlossaryPullOptions {
+
+    private String fileType;
+    private String[] transLang;
+
+    @Option(name = "--file-type", metaVar = "(CSV or PO)",
+        usage = "File type to be downloaded.\n" +
+            "csv - csv file format with comma separated\n" +
```
Comment (djansen-redhat) [inline zanata-client-commands/src/main/java/org/zanata/client/commands/glossary/pull/GlossaryPullOptionsImpl.java]: (default)

**Comment by djansen-redhat** [comment]
- [ ] Batch size default no work

**Comment by djansen-redhat** [comment]
- [ ] Bad locale identifier results in NPE http://pastebin.com/ScdUE6gM

**Comment by aeng** [comment]
@djansen-redhat since web-upload only supports `en-US` as the source language, I've updated the client to do the same. So --src-lang is removed.

**Comment by djansen-redhat** [comment]
:white_check_mark: verified

**Pull request merged**